packages/language-service/ivy/test/type_definitions_spec.ts

```diff
@@ -121,7 +125,11 @@ describe('type definitions', () => {
         const definitions = getTypeDefinitionsAndAssertBoundSpan({
           templateOverride: `<div *ngFor="let item o¦f heroes"></div>`,
         });
-        expectAllArrayDefinitions(definitions);
+        // In addition to all the array defs, this will also return the NgForOf def because the
+        // input is part of the selector ([ngFor][ngForOf]).
+        expectAllDefinitions(
+            definitions, new Set(['Array', 'NgForOf']),
+            possibleArrayDefFiles.add('ng_for_of.d.ts'));
```

Comment:
This should not mutate `possibleArrayDefFiles`:

```suggestion
            new Set(possibleArrayDefFiles).add('ng_for_of.d.ts'));
```

or alternatively:
```suggestion
            new Set([...possibleArrayDefFiles, 'ng_for_of.d.ts']);
```